src/ConfigTrait.php

```diff
@@ -54,33 +52,37 @@ public function readConfig($files = ['config.php'], $format = 'php')
             if (!is_readable($file)) {
                 throw new Exception(['Can not read config file', 'file' => $file, 'format' => $format]);
             }
-
+    
+            $tempConfig = [];
+            
             switch (strtolower($format)) {
                 case 'php':
                     $config = null;
                     require $file; // fills $config variable
-                    $this->config = $config;
+                    $tempConfig = $config;
                     break;
 
                 case 'php-inline':
-                    $this->config = require $file;
+                    $tempConfig = require $file;
                     break;
 
                 case 'json':
-                    $this->config = json_decode(file_get_contents($file), true);
+                    $tempConfig = json_decode(file_get_contents($file), true);
                     break;
 
                 case 'yaml':
                     // @codeCoverageIgnoreStart
                     if (!class_exists(\Symfony\Component\Yaml\Yaml::class)) {
                         throw new Exception(['You need Symfony\Yaml repository if you want to parse YAML files']);
                     }
-                    $this->config = \Symfony\Component\Yaml\Yaml::parseFile($file);
+                    $tempConfig = \Symfony\Component\Yaml\Yaml::parseFile($file);
                     // @codeCoverageIgnoreEnd
                     break;
             }
+            
+            $this->config = array_merge($this->config, $tempConfig);
```

Comment:
`array_merge` or `array_merge_recursive` here?
keep in mind that config options can have deep structure too so i think array_merge_recursive would be better.